script/kibana-migration.py

```diff
@@ -0,0 +1,121 @@
+
+    print "Postfix all ids with -7x"
+    ids = get_replacable_ids()
+    rename_entries(ids)
+
+    print "Postfix all titles with ` 7.x`"
+    titles = get_replacable_titles()
+    rename_entries(titles)
+
+
+def get_replacable_ids():
+    files = get_files()
+
+    ids = {}
+    for file in files:
+        with open(file, 'r') as f:
+            objects = yaml.safe_load(f)
+
+            for v in objects["objects"]:
+                # Checks if an id was already migrated, if not adds it to the list
+                if "-7x" not in v["id"]:
+                    ids[v["id"]] = v["id"] + "-7x"
+
+    return ids
+
+
+def read_migration_fields():
+    migration_fields = {}
+    migration_yml = "../dev-tools/ecs-migration.yml"
+    with open(migration_yml, 'r') as f:
+        migration = yaml.safe_load(f)
+        for k in migration:
+            if "to" in k and "from" in k:
+                if isinstance(k["to"], basestring):
+                    # Add "{}" around fields to make them more unique and not have false positives
+                    migration_fields['"' + k["from"] + '"'] = '"' + k["to"] + '"'
+                    # Some fields exist inside a query / filter where they are followed by :
+                    migration_fields[k["from"] + ':'] = k["to"] + ':'
+
+    return migration_fields
+
+
+def get_replacable_titles():
+    files = get_files()
+
+    titles = {}
+    for file in files:
+        with open(file, 'r') as f:
+            objects = yaml.safe_load(f)
+
+            for v in objects["objects"]:
+
+                # Add "{}" around titles to make them more unique and not have false positives
+                if "title" in v["attributes"]:
+                    if "7.x" not in v["attributes"]["title"]:
+                        titles['"' + v["attributes"]["title"] + '"'] = '"' + v["attributes"]["title"] + " 7.x" + '"'
+
+                if "visState" in v["attributes"] and "title" in v["attributes"]["visState"]:
+                    if "7.x" not in v["attributes"]["visState"]["title"]:
+                        titles['"' + v["attributes"]["visState"]["title"] + '"'] = '"' + \
+                            v["attributes"]["visState"]["title"] + " 7.x" + '"'
+
+    return titles
+
+
+def rename_entries(renames):
+    files = get_files()
+
+    for file in files:
+        print file
+        s = open(file).read()
+        for k in renames:
+            s = s.replace(k, renames[k])
+        f = open(file, 'w')
+        f.write(s)
+        f.close()
+
+
+def get_files():
+    all_beats = '../*/_meta/kibana/6/dashboard/*.json'
+    module_beats = '../*/module/*/_meta/kibana/6/dashboard/*.json'
+    heartbeat = '../hearbeat/monitors/active/*/_meta/kibana/6/dashboard/*.json'
+    xpack_module_beats = '..x-pack/*/module/*/_meta/kibana/6/dashboard/*.json'
+
+    return glob.glob(all_beats) + glob.glob(module_beats) + glob.glob(heartbeat) + glob.glob(xpack_module_beats)
+
+
+if __name__ == "__main__":
+    migration()
```

Comment:
Would it make sense to have a "reverse" mode for backports?

Reply:
Reverse it not really feasible for the fields migration as there are many to 1 mappings.

Reply:
Ok, if we have to backport something in dashboards we can do it by hand if it involves renamed fields.